sdk/src/loom/Utils/Injector.ls

```diff
@@ -105,31 +139,69 @@ class Injector
     public function apply(target:Object, tagName:String="Inject"):void
     {
         var type = target.getType();
-        var targets:Vector.<InjectionTarget> = injectionTargetCache[type];
 
+        // Get all potential targets from the cache.
+        var targets:Vector.<InjectionTarget> = injectionTargetCache[type];
         if(!targets)
         {
+            // Cache miss, create entry.
             targets = new Vector.<InjectionTarget>();
             injectionTargetCache[type] = targets;
 
-            // otherwise we need to create injection targets
+            // We need to create injection targets...
+
+            // ...for fields...
             for(var i = 0; i<type.getFieldInfoCount(); i++)
             {
+                // Look up the field.
                 var fieldInfo = type.getFieldInfo(i);
+
+                // Get the metadata if present.
                 var inject = fieldInfo.getMetaInfo(tagName);
-                if(inject) {
-                    var t = new InjectionTarget();
-                    t.type = fieldInfo.getTypeInfo();
-                    t.property = fieldInfo.getName();
-                    t.fieldInfo = fieldInfo;
-                    t.id = inject.getAttribute("id");
+                if(!inject)
+                    continue;
 
-                    if(t.id == null)
-                        t.id = fieldInfo.getName();
+                // Create the injection target.
+                var t = new InjectionTarget();
+                t.type = fieldInfo.getTypeInfo();
+                t.property = fieldInfo.getName();
+                t.fieldInfo = fieldInfo;
+                t.id = inject.getAttribute("id");
 
-                    targets.push(t);
-                }
+                if(t.id == null)
+                    t.id = fieldInfo.getName();
+
+                targets.push(t);
             }
+
+            // ...for properties...
+            for(i = 0; i<type.getPropertyInfoCount(); i++)
+            {
+                // Look up the property.
+                var propInfo = type.getPropertyInfo(i);
+
+                // Get first of getter, setter for the metadata.
+                inject = null;
+                if(propInfo.getGetMethod())
+                    inject = propInfo.getGetMethod().getMetaInfo(tagName);
+                if(!inject && propInfo.getSetMethod())
```

Comment:
It falls through. So if it can't find the get, it tries the set (if inject isn't set). Should I add a comment about that?